waffle_hub/dataset/adapter/autocare_dlt.py

```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+from typing import Union
+
+from waffle_utils.file import io
+
+from waffle_hub import TaskType
+from waffle_hub.utils.conversion import convert_rle_to_polygon
+
+
+def _export_autocare_dlt(
+    self,
+    export_dir: Path,
+    train_ids: list,
+    val_ids: list,
+    test_ids: list,
+    unlabeled_ids: list,
+):
+    """Export dataset to Autocare DLT format
+
+    Args:
+        export_dir (Path): Path to export directory
+        train_ids (list): List of train ids
+        val_ids (list): List of validation ids
+        test_ids (list): List of test ids
+        unlabeled_ids (list): List of unlabeled ids
+    """
+    io.make_directory(export_dir)
+
+    image_dir = export_dir / "images"
+
+    for split, image_ids in zip(
+        ["train", "val", "test", "unlabeled"],
+        [train_ids, val_ids, test_ids, unlabeled_ids],
+    ):
+        if len(image_ids) == 0:
+            continue
+
+        coco = {
+            "categories": [
+                {
+                    "id": category.category_id,
+                    "name": category.name,
+                    "supercategory": category.supercategory,
+                }
+                for category in self.categories.values()
+            ],
+            "images": [],
+            "annotations": [],
+        }
+        category_names = [category.name for category in self.categories.values()]
+
+        for image_id in image_ids:
+            image = self.images[image_id]
+            image_path = self.raw_image_dir / image.file_name
+            image_dst_path = image_dir / image.file_name
+            io.copy_file(image_path, image_dst_path, create_directory=True)
+
+            d = image.to_dict()
+            image_id = d.pop("image_id")
+            coco["images"].append({"id": image_id, **d})
+
+            annotations = self.image_to_annotations[image_id]
+            for annotation in annotations:
+                d = annotation.to_dict()
+                if d.get("segmentation", None):
+                    if isinstance(d["segmentation"], dict):
+                        d["segmentation"] = convert_rle_to_polygon(d["segmentation"])
+                caption = d.get("caption", None)
+                if caption:
+                    for c in set(caption):
+                        if c not in category_names:
+                            raise ValueError(
+                                f"character {c} is not in categories, please check your dataset"
+                            )
```

Comment:
I think it would be better that the category check should be done when adding annotations to Dataset.

Reply:
> I think it would be better that the category check should be done when adding annotations to Dataset.

I've added two places, but would it be better to exclude this one?